src/nncf/quantization/algorithms/weight_compression/algorithm.py

```diff
@@ -744,106 +791,111 @@ def apply(
 
         nodes_to_compress = self.get_nodes_to_compress(graph)
 
-        statistics = None
-        if (self._data_aware_mixed_precision or self._data_aware_compression) and dataset:
-            matmul_nodes_to_compress = [
-                node for node in nodes_to_compress if node.metatype in self._backend_entity.matmul_metatypes
-            ]
-            matmul_input_to_output_nodes_map = self.get_matmul_input_to_output_nodes_map(
-                matmul_nodes_to_compress, graph
-            )
-            if statistic_points is None:
-                statistic_points = self.get_statistic_points(model, graph, matmul_input_to_output_nodes_map.keys())
-                statistic_points = self._collect_statistics(dataset, graph, model, statistic_points)
-            statistics = self._get_statistics_for_weights_compression(
-                matmul_input_to_output_nodes_map, statistic_points
-            )
-
         all_weight_params: list[WeightCompressionParameters] = []
-        weight_names = set()
+        skipped_weight_params: list[WeightCompressionParameters] = []
 
-        is_last_layer_shared = False
+        weight_names = set()
+        is_last_layer_skipped = False
         n = len(nodes_to_compress)
+        ignored_names = self.get_ignored_node_names(graph)
         for i, node in enumerate(nodes_to_compress):
+            is_target_node = should_consider_scope(node.node_name, ignored_names)
             for weight_name, weight_port_id in self._backend_entity.get_weight_names_and_port_ids(node, graph):
+                is_last_layer = i == n - 1
                 if weight_name in weight_names:
-                    if i == n - 1:
-                        is_last_layer_shared = True
+                    is_last_layer_skipped = is_last_layer
```

Comment:
I have a concern that previously this variable was named `is_last_layer_shared` and from that it was clear why the node is skipped. Now it is less obvious. Perhaps, some comment would help.